network/p2p/scoring/app_score_test.go

```diff
@@ -111,37 +114,20 @@ func TestFullGossipSubConnectivity(t *testing.T) {
 // the network of honest nodes.
 func TestFullGossipSubConnectivityAmongHonestNodesWithMaliciousMajority(t *testing.T) {
 	// Note: if this test is ever flaky, this means a bug in our scoring system. Please escalate to the team instead of skipping.
-	total := 10
-	for i := 0; i < total; i++ {
-		if !testGossipSubMessageDeliveryUnderNetworkPartition(t, true) {
-			// even one failure should not happen, as it means that malicious majority can partition the network
-			// with our peer scoring parameters.
-			require.Fail(t, "honest nodes could not exchange message on GossipSub")
-		}
+	if !testGossipConnectivityUnderNetworkPartition(t, true) {
+		// even one failure should not happen, as it means that malicious majority can partition the network
+		// with our peer scoring parameters.
+		require.Fail(t, "honest nodes are not on each others' topic mesh on GossipSub")
 	}
-}
 
-// TestNetworkPartitionWithNoHonestPeerScoringInFullTopology is part one of testing pushing access nodes to the edges of the network.
-// This test proves that if access nodes are NOT pushed to the edge of network, a malicious majority of them can
-// partition the network by disconnecting honest nodes from each other even when the network topology is a complete graph (i.e., full topology).
-func TestNetworkPartitionWithNoHonestPeerScoringInFullTopology(t *testing.T) {
-	unittest.SkipUnless(t, unittest.TEST_FLAKY, "to be fixed later")
-	total := 100
-	for i := 0; i < total; i++ {
-		// false means no honest peer scoring.
-		if !testGossipSubMessageDeliveryUnderNetworkPartition(t, false) {
-			return // partition is successful
-		}
-	}
-	require.Fail(t, "expected at least one network partition")
 }
 
-// testGossipSubMessageDeliveryUnderNetworkPartition tests that whether two honest nodes can exchange messages on GossipSub
+// testGossipConnectivityUnderNetworkPartition tests that whether two honest nodes are in each others topic mesh on GossipSub
 // when the network topology is a complete graph (i.e., full topology) and a malicious majority of access nodes are present.
 // If honestPeerScoring is true, then the honest nodes are enabled with peer scoring.
-// A true return value means that the two honest nodes can exchange messages.
-// A false return value means that the two honest nodes cannot exchange messages within the given timeout.
-func testGossipSubMessageDeliveryUnderNetworkPartition(t *testing.T, honestPeerScoring bool) bool {
+// A true return value means that the two honest nodes have each other in each others' mesh.
```

Comment:
```suggestion
// A true return value means that the two honest nodes have each other in each other's mesh.
```

Reply:
Also suggestion for line 129:
`// A false return value means that the two honest nodes do not have each other in each other's mesh (at least one of them is partitioned).`